clients/client-py/taskcluster/helper.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+import logging
+from taskcluster.generated import _client_importer
+from taskcluster.generated.aio import _client_importer as _async_client_importer
+
+logger = logging.getLogger(__name__)
+
+
+class TaskclusterConfig(object):
+    """
+    Local configuration used to access Taskcluster service and objects
+    """
+
+    def __init__(self, url=None):
+        self.options = None
+        self.secrets = None
+        self.default_url = os.environ.get("TASKCLUSTER_ROOT_URL", url)
+        assert self.default_url is not None, "You must specify a Taskcluster deployment url"
+
+    def auth(self, client_id=None, access_token=None, max_retries=12):
+        """
+        Build Taskcluster credentials options
+        Supports, by order of preference:
+         * directly provided credentials
+         * credentials from environment variables
+         * taskclusterProxy
+         * no authentication
+        """
+        self.options = {"maxRetries": max_retries}
+
+        if client_id is None and access_token is None:
+            # Credentials preference: Use env. variables
+            client_id = os.environ.get("TASKCLUSTER_CLIENT_ID")
+            access_token = os.environ.get("TASKCLUSTER_ACCESS_TOKEN")
+            logger.info("Using taskcluster credentials from environment")
+        else:
+            logger.info("Using taskcluster credentials from cli")
+
+        if client_id is not None and access_token is not None:
+            # Use provided credentials
+            self.options["credentials"] = {
+                "clientId": client_id,
+                "accessToken": access_token,
+            }
+            self.options["rootUrl"] = self.default_url
+
+        elif "TASK_ID" in os.environ:
+            # Use Taskcluster Proxy when running in a task
+            logger.info("Taskcluster Proxy enabled")
+            self.options["rootUrl"] = "http://taskcluster"
```

Comment:
Should this be `os.environ.get("TASKCLUSTER_PROXY_URL")`? https://community-tc.services.mozilla.com/docs/manual/design/env-vars

Reply:
Good catch ! I'll update the PR